test/integration/test_plugin.cpp

```diff
@@ -9,25 +9,191 @@
 
 #include "gtest/gtest.h"
 
-int plugin_test_counter_pre{0};
-int plugin_test_counter_post{0};
+int plugin_test_capture_counter_pre{0};
```

Comment:
Can you make a test where the capture and launch counters do not end up being the same?

Reply:
Given the way that the capture and launch counters are implemented in this PR I think they should always be equal.
This is something I wasn't sure about though, for example when using IndexSet should the plugins be called once or once per segment?

Reply:
I think the only cases where they would not be the same would be in your work group stuff or async stuff. So I think this OK for this PR. I think the lambda would have to be captured per segment launch since it uses the forall impl stuff, correct?

Reply:
The IndexSet iteration happens in the wrap::forall level which is one level below where the plugins are being called so the lambda has already been captured, and the while wrap::forall index set implementation is part of the launch.

Reply:
OK. Makes sense.